botcore/utils/channel.py

```diff
@@ -0,0 +1,35 @@
+"""Useful helper functions for interacting with various discord.py channel objects."""
+
+import discord
+from discord.ext.commands import Bot
+
+from botcore.utils import loggers
+
+log = loggers.get_logger(__name__)
+
+
+def is_in_category(channel: discord.TextChannel, category_id: int) -> bool:
+    """
+    Return whether the given channel has the category_id.
+
+    Args:
+        channel: The channel to check.
+        category_id: The category to check for.
+
+    Returns:
+        A bool depending on whether the channel is in the category.
+    """
+    return getattr(channel, "category_id", None) == category_id
+
+
+async def get_or_fetch_channel(bot: Bot, channel_id: int) -> discord.abc.GuildChannel:
+    """Attempt to get or fetch a channel and return it."""
```

Comment:
This could use a better docstring that documents its parameters.

Reply:
https://github.com/python-discord/bot-core/pull/29/commits/d7841816581737b4a7cece5bba98328a44630d4b